### DISCUSSION TITLE
Path(alias=) feature doesn't work if the alias has a hyphen in it

### DISCUSSION THREAD

**Discussion by jrobbins-LiveData** [fastapi/fastapi#6188] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
This route, based on Closed Issue https://github.com/tiangolo/fastapi/issues/131#issuecomment-1177187775, works, and shows how one can alias a path parameter.


@router.get("/api/users/{id}/profile/")
def user_profile(user_id: int = Path(None, alias="id")):
    return {
        "id": user_id,
        "username": "johndoe"
    }
```

This route, using `"user-id"` instead of `"id"` as the alias, does not work.

```
@router.get("/api/users1/{user-id}/profile/")
def user_profile(user_id: int = Path(None, alias="user-id")):
    return {
        "id": user_id,
        "username": "johndoe"
    }
```
```


### Description

Both `Query()` and `Path()` have an `alias` feature.  `Query()` aliases can contain hyphens, but `Path()` aliases cannot.  This appears to be a bug, as the [doc](https://fastapi.tiangolo.com/tutorial/path-params-numeric-validations/#declare-metadata) says about Path metadata

> You can declare all the same parameters as for Query.

### Operating System

Linux

### Operating System Details

Amazon Linux 2 

### FastAPI Version

0.86.0

### Python Version

3.9.13

### Additional Context

Swagger UI showing working route:
<img width="801" alt="path_alias_working" src="https://user-images.githubusercontent.com/79278744/201563302-a0539e26-af29-4785-b2f5-ec970f3233f1.png">

Swagger UI showing hyphenated alias Not Found:
<img width="803" alt="page_alias_with_hyphen_issue" src="https://user-images.githubusercontent.com/79278744/201563374-dc3f2198-ca49-42eb-974b-557f3a3a9559.png">

**Comment by JarroVGIT** [comment]
Duplicate of https://github.com/tiangolo/fastapi/issues/5281?

**Comment by jrobbins-LiveData** [comment]
Yes, appears to be a duplicate. (#5281 misspells "hyphens" as "hypens".)

**Comment by jrobbins-LiveData** [comment, marked as answer]
Closing since this is a duplicate of #5281.

**Comment by tiangolo** [comment]
Thanks for the help here @JarroVGIT ! 🚀

I'm gonna go ahead and close this then. 🤓
